code-reviews.md

```diff
@@ -0,0 +1,80 @@
+# Code reviews at FUN
+
+Code reviews are collaborative steps in the development of our projects.
+The purpose of code reviews is twofold. They allow both to:
+* check the quality of the code and
+* to communicate actively on the evolutions.
+
+Code reviews concern all members who are involved in the development team, regardless of their competence level.
+This concept raises questions about their effectiveness and relevance both on the side of the developer submitting the review and on the side of the developers reviewing the code.
+
+> When ? How? How long do we review codes?
+> Which focus? Which feedback to give to the reviewer?
+
+## Approach of code reviews
+
+Repository managers such as GitHub or GitLab are helpful to review codes.
+Associated workflows, pull request \(PR\) or merge request \(MR\) ease the tracking of information shared in code reviews.
+
+We recommend considering a fine granularity of code reviews.
+The benefit is both temporal and qualitative.
+By doing code reviews as frequently as possible, the exercise is made easier and all members feel more involved in development.
+By doing light and targeted code reviews on a specific topic, the review is done with more attention because it is less complex.
+
+A code review is an opportunity to **actively** communicate about the changes to other developers and get feedback on the quality of the changes.
+As far as possible, a code review should be done in small changes.
+In this perspective, we advise against doing code reviews for a collection of changes that can be analyzed independently.
+However, it is important to keep in mind that what is under review must be understandable in its software context.
+Some complex refactorings or features can not be split in small changes as they would lose software design information.
+The reviewer would then fail to comment on design choices and be limited to a simple technical review.
+
+When opening a pull request \(PR\) or merge request \(MR\) for a review, we recommend to assess the granularity of the review to keep visible the algorithm and design of the feature.
+
+## Best practices for reviewing codes
+
+The challenge of a code review is to position oneself in the face of another member's work production.
+It is not a question of sanctioning his/her work, but of assessing it; issuing an opinion.
+
+Here are 5 good practices to adopt that may be beneficial for a code review.
+
+1. Engage conversation for clarification 💬
+
+The point here is to question what has been produced.
+It is not because the reviewer has difficulty understanding the code that the code quality is bad.
+Code reviews is an exercise that enhances the value of the work produced in a team spirit.
+We recommend to asking the developer why he/she did it rather than being prescriptive and tell developer to code in a specific way.

```

Comment:
I think it's recommend + infinitive but not sure?

Reply:
Yes, it's "we recommend to ask" or "we recommend asking" but definitely not both of them ...

Reply:
I knew something was wrong! You nailed it ^^

Reply:
Just to contribute :-), or you could say "we recommend on asking"